imageprocessor/imageprocessor-core/src/main/java/pl/edu/uj/kimage/processingFlow/ImageProcessingFlow.java

```diff
@@ -2,30 +2,47 @@
 
 import pl.edu.uj.kimage.Image;
 import pl.edu.uj.kimage.eventbus.Event;
+import pl.edu.uj.kimage.eventbus.EventBus;
+import pl.edu.uj.kimage.plugin.FlowEvent;
+import pl.edu.uj.kimage.plugin.FlowStep;
+import pl.edu.uj.kimage.plugin.StepResultEvent;
 
 import java.util.Collection;
-import java.util.concurrent.atomic.AtomicInteger;
 
 public class ImageProcessingFlow {
     private final FlowStepRepository flowStepRepository;
-    private AtomicInteger currentStep;
+    private final EventBus eventBus;
+    private final int flowLength;
 
-    //TODO add processing mechanism, event bus
-
-    public ImageProcessingFlow(FlowStepRepository flowStepRepository) {
+    ImageProcessingFlow(FlowStepRepository flowStepRepository, EventBus eventBus, int flowLength) {
         this.flowStepRepository = flowStepRepository;
-        this.currentStep = new AtomicInteger(0);
+        this.eventBus = eventBus;
+        this.flowLength = flowLength;
+    }
+
+    public void start(Image image) {
+        registerFlowToEventBus();
+        StepResultEvent<Image> integerStepResultEvent = new StepResultEvent<>(0, image);
+        eventBus.publish(integerStepResultEvent);
     }
 
-    public void processEvent(Event event){
-        Collection<FlowStep> notProcessedSteps = flowStepRepository.loadAllLaterThen(currentStep.get());
-        notProcessedSteps.forEach(e->e.processEvent(event));
+    private void processEvent(FlowEvent event) {
+        Collection<FlowStep> notProcessedSteps = flowStepRepository.loadAll();
+        notProcessedSteps.forEach(e -> e.processEvent(event));
     }
 
-    public void start(Image image)
-    {
-        StepResultEvent<Integer> integerStepResultEvent = new StepResultEvent<>(0, 1);
+    private void registerFlowToEventBus() {
+        eventBus.registerEventListener(StepResultEvent.class, (event -> {
+            if (event.getStepId() == flowLength) {
```

Comment:
Why do you compare the step identifier with length of the process? Why did you assume that the last step identifier will be equal to the length (amount of steps) of the process?